pkg/drivers/drivers.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drivers
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/cloudflare/cfssl/log"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/mcnflag"
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/docker/machine/libmachine/ssh"
+	"github.com/pkg/errors"
+)
+
+func GetDiskPath(d *drivers.BaseDriver) string {
+	return filepath.Join(d.ResolveStorePath("."), d.GetMachineName()+".rawdisk")
+}
+
+type CommonDriver struct{}
+
+//Not implemented yet
+func (d *CommonDriver) GetCreateFlags() []mcnflag.Flag {
+	return nil
+}
+
+//Not implemented yet
+func (d *CommonDriver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	return nil
+}
+
+func createRawDiskImage(sshKeyPath, diskPath string, diskSizeMb int) error {
+	tarBuf, err := mcnutils.MakeDiskImage(sshKeyPath)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(diskPath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	file.Seek(0, os.SEEK_SET)
```

Comment:
Do you happen to know if we actually need this call? It never made sense to me that we would.

Reply:
I think its in case the file already exists and has contents.  The magic boot2docker header that we write in the next lines needs to be the first bytes